src/App.tsx

```diff
@@ -14,11 +14,92 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  selectedGood: string[];
+};
+
+class App extends React.Component<{}, State> {
+  state = {
+    selectedGood: ['Jam'],
+  };
+
+  buttonClick = (item: string) => {
+    const { selectedGood } = this.state;
+
+    if (selectedGood.includes(item)) {
+      this.setState({ selectedGood: selectedGood.filter(e => e !== item) });
+    } else {
+      this.setState({ selectedGood: [...selectedGood, item] });
+    }
+  };
+
+  clearGoods = () => {
+    this.setState({
+      selectedGood: [],
+    });
+  };
+
+  render() {
+    const { selectedGood } = this.state;
+    let title = '';
+
+    switch (selectedGood.length) {
+      case 0:
+        title = 'No goods selected';
+        break;
+      case 1:
+        title = `${selectedGood} is selected`;
+        break;
+
+      case 2:
+        title = `${selectedGood[0]} and ${selectedGood[1]} are selected`;
+        break;
+      default:
+        title = `${selectedGood.slice(0, selectedGood.length - 1).join(',')} and ${selectedGood[selectedGood.length - 1]} are selected`;
```

Comment:
можна юзати метод `at`
```suggestion
        title = `${selectedGood.slice(0, selectedGood.length - 1).join(',')} and ${selectedGood.at(-1)} are selected`;
```